### ISSUE TITLE
linemode fileinfo[Mi] crashes ranger with python3

### ISSUE THREAD

**Issue by hillyu** [ranger/ranger#992]
#### ISSUE TYPE

- Bug report
`ranger version: 1.8.1, executed with python 3.6.3
Locale: en_US.UTF-8
Current file: /home/hill/bin
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/ranger/core/main.py", line 145, in main
    fm.loop()
  File "/usr/lib/python3.6/site-packages/ranger/core/fm.py", line 362, in loop
    ui.redraw()
  File "/usr/lib/python3.6/site-packages/ranger/gui/ui.py", line 295, in redraw
    self.draw()
  File "/usr/lib/python3.6/site-packages/ranger/gui/ui.py", line 322, in draw
    DisplayableContainer.draw(self)
  File "/usr/lib/python3.6/site-packages/ranger/gui/displayable.py", line 248, in draw
    displayable.draw()
  File "/usr/lib/python3.6/site-packages/ranger/gui/widgets/view_miller.py", line 99, in draw
    DisplayableContainer.draw(self)
  File "/usr/lib/python3.6/site-packages/ranger/gui/displayable.py", line 248, in draw
    displayable.draw()
  File "/usr/lib/python3.6/site-packages/ranger/gui/widgets/browsercolumn.py", line 168, in draw
    self._draw_directory()
  File "/usr/lib/python3.6/site-packages/ranger/gui/widgets/browsercolumn.py", line 348, in _draw_directory
    infostringdata = current_linemode.infostring(drawn, metadata)
  File "/usr/lib/python3.6/site-packages/ranger/core/linemode.py", line 107, in infostring
    fileinfo = fileinfo.decode("utf-8")
AttributeError: 'str' object has no attribute 'decode'
`

**Comment by Vifon** [comment]
Could you retest on the master branch directly from Github? I believe it was already fixed but I'd like to be sure.

**Comment by three-comrades** [comment]
I get this with the current master and Python 3.6.4:

```
> ranger -c
ranger version: 1.6.1, executed with python 3.6.4
Locale: en_US.UTF-8
Current file: /home/kelvin/tmp/ranger/build
Traceback (most recent call last):
  File "/home/kelvin/tmp/ranger/venv/lib/python3.6/site-packages/ranger/core/main.py", line 139, in main
    fm.loop()
  File "/home/kelvin/tmp/ranger/venv/lib/python3.6/site-packages/ranger/core/fm.py", line 319, in loop
    ui.redraw()
  File "/home/kelvin/tmp/ranger/venv/lib/python3.6/site-packages/ranger/gui/ui.py", line 267, in redraw
    self.draw()
  File "/home/kelvin/tmp/ranger/venv/lib/python3.6/site-packages/ranger/gui/ui.py", line 293, in draw
    DisplayableContainer.draw(self)
  File "/home/kelvin/tmp/ranger/venv/lib/python3.6/site-packages/ranger/gui/displayable.py", line 246, in draw
    displayable.draw()
  File "/home/kelvin/tmp/ranger/venv/lib/python3.6/site-packages/ranger/gui/widgets/browserview.py", line 98, in draw
    DisplayableContainer.draw(self)
  File "/home/kelvin/tmp/ranger/venv/lib/python3.6/site-packages/ranger/gui/displayable.py", line 246, in draw
    displayable.draw()
  File "/home/kelvin/tmp/ranger/venv/lib/python3.6/site-packages/ranger/gui/widgets/browsercolumn.py", line 164, in draw
    self._draw_directory()
  File "/home/kelvin/tmp/ranger/venv/lib/python3.6/site-packages/ranger/gui/widgets/browsercolumn.py", line 326, in _draw_directory
    textstring = self._draw_text_display(text, space)
UnboundLocalError: local variable 'text' referenced before assignment

ranger crashed.  Please report this traceback at:
https://github.com/hut/ranger/issues
```

**Comment by three-comrades** [comment]
Sorry, I accidentally deleted my last comment instead of editing it. This seems to be fixed on master with Python 3.6.4. The crash I observed was due to misconfiguration on my part ...

**Comment by toonn** [comment]
Thanks for testing.
